datapipelines/src/main/kotlin/io/customer/sdk/android/CustomerIO.kt

```diff
@@ -157,11 +186,70 @@ class CustomerIO private constructor(
     }
 
     override fun clearIdentify() {
-        val userId = registeredUserId ?: "anonymous"
-        logger.debug("resetting user profile with id $userId")
+        logger.info("resetting user profile with id $registeredUserId")
+
+        logger.debug("deleting device token to remove device from user profile")
+        deleteDeviceToken()
+
+        logger.debug("resetting user profile")
         analytics.reset()
     }
 
+    override val registeredDeviceToken: String?
+        get() = globalPreferenceStore.getDeviceToken()
+
+    override var deviceAttributes: CustomAttributes
+        get() = emptyMap()
```

Comment:
if possible lets check if analytics is holding these in memory/disk and we can just return those?

Reply:
If analytics is holding it somewhere, it will be good to return, but doesn't look like so. So let's stick to this for now.